In [1]:
import cmath
import numpy as np
import math

def calc_parallel(R1,R2):
    return R1*R2/(R1+R2)

def V_div(Vs,R1,R2):
    return Vs*R1/(R1+R2)

def I_div(Is,R1,R2):
    return Is*R2/(R1+R2)

def calc_exp(A,ab,t):
    return A*np.exp(-1*ab*t)

_____
### Clase Práctica 4

#### Temas
* Capacitores e inductores.
* Circuitos en estado transitorio.

#### Bibliografía: Análisis de circuitos en ingeniería. Hayt $et \: al.$ 8va ed. Capítulo 7, 8 y 9

___
#### En el capacitor:

$i(t)=C\dfrac{dv}{dt}$

#### En el inductor:
$v(t)=L\dfrac{di}{dt}$

#### Ecuación característica:

#### $a\dfrac{df}{dt}+bf=0$ tiene como solución $f=Ae^{-bt/a}$

_____
#### 1- El interruptor del circuito en la figura ha estado cerrado durante un tiempo ridículamente largo hasta abrirse de manera repentina en $t = 0$. 
 
#### a) Obtenga expresiones para $i_L$ y $v$ en el circuito de la figura que sean válidas para todo $t \geq 0$. 
 
#### b) Calcule $i_L(t)$ y $v(t)$ en el instante inmediatamente anterior a la apertura del interruptor, en el instante inmediatamente posterior a la apertura del interruptor, y en el $t = 470 \: \mu s$.

<img src="c1.png" alt= “” width="40%" height="40%">

#### a) para $t>0:$

$25i_L+10i_L+40*10^{-3}\dfrac{di_L}{dt}+50i_L=0$

$4*10^{-2}\dfrac{di_L}{dt}+85i_L=0$

#### Utilizando la ecuación característica:

$i_L=I_0e^{\dfrac{-8500t}{4}} \: \mathrm{[A]}$ 

#### Calcular $I_0$ en $t=0^-$:

$i_L=\dfrac{10}{50}=0.2 \: \mathrm{A}$

#### Entonces para $t>0:$

$i_L=0.2e^{\dfrac{-8500t}{4}} \: \mathrm{[A]}$ 

$v=25i_L=5e^{\dfrac{-8500t}{4}} \: \mathrm{[V]}$

In [2]:
print('t= 0-','i_L:',10/50,'A','v:',V_div(10,25,10),'V')
print('t= 0+','i_L:',calc_exp(0.2,8500/4,0),'A','v:',calc_exp(5,8500/4,0),'V')
print('t= 470 us','i_L:',1e3*calc_exp(0.2,8500/4,470*1e-6),'mA','v:',calc_exp(5,8500/4,470*1e-6),'V')

t= 0- i_L: 0.2 A v: 7.142857142857143 V
t= 0+ i_L: 0.2 A v: 5.0 V
t= 470 us i_L: 73.66791559970198 mA v: 1.8416978899925496 V


___
#### 2- Es seguro suponer que un interruptor dibujado en el circuito de la figura ha estado cerrado durante un tiempo tan largo que cualquier transitorio que podría haber surgido desde la primera vez que se conectó la fuente de tensión ha desaparecido.
 
#### a) Determine la constante de tiempo del circuito. 
 
#### b) Calcule la tensión $v(t)$ en $t =\tau$, $2\tau$ y $5\tau$ .

<img src="c2.png" alt= “” width="40%" height="40%">

#### a) En $t=0^+$:

$V_0=4 \: \mathrm{V}$

$2*10^{-9}\dfrac{dv}{dt}+\dfrac{v}{200+100+150}=0$

$v=4e^{\dfrac{-t}{2*10^{-9}*450}} \: \mathrm{[V]}$

$\tau=RC=9*10^{-7}$

In [3]:
tau=9e-7
print('v(tau):',calc_exp(4,1/tau,tau),'V')
print('v(2tau):',calc_exp(4,1/tau,2*tau),'V')
print('v(5tau):',calc_exp(4,1/tau,5*tau),'V')

v(tau): 1.4715177646857693 V
v(2tau): 0.5413411329464508 V
v(5tau): 0.026951787996341844 V


___
#### 3- Muestre en el circuito de la figura que para una fuente senoidal la relación entre los fasores de voltaje y corriente en el inductor es $\mathrm{j}2\pi fL$ y en el capacitor es $\dfrac{-\mathrm{j}}{2\pi fC}$.

<img src="c3_1.png" alt= “” width="40%" height="40%">

$v_s(t)= Ri(t)+L\dfrac{di}{dt}+\dfrac{1}{C}\int\limits^{t}_{t_0} i(t')dt'+v(t_0)$

#### entonces:

$i(t)=I\cos(2\pi ft+\phi)$

#### Añadiendo una fuente compleja por superposición:

$i(t)=Ie^{j(2\pi ft+\phi)}$

$v_s(t)= RIe^{j(2\pi ft+\phi)}+L\dfrac{d(Ie^{j(2\pi ft+\phi)})}{dt}+\dfrac{1}{C}\int\limits^{t}_{t_0} Ie^{j(2\pi ft'+\phi)}dt'+v(t_0)$

$v_s(t)= RIe^{j(2\pi ft+\phi)}+j2\pi L I e^{j(2\pi ft+\phi)}+\dfrac{1}{j2\pi C}Ie^{j(2\pi ft'+\phi)}$

#### Hacer $V/I$ en cada componente

___
#### 4- Diga el fasor o la expresión temporal según corresponda:

2- a) $i_1(t)=12\sin(400t+110^{\circ}) \: [\mathrm{A}]$

**$\sin(x)=\cos(x-90)$

In [4]:
ph_a=(12,20)
print('resultado: ',ph_a)

resultado:  (12, 20)


b) $i_2(t)=-7\sin(200t)- 3\cos(200t - 30^{\circ}) \: [\mathrm{A}]$

In [5]:
ph_b1=cmath.rect(7,math.radians(90))
ph_b2=cmath.rect(3,math.radians(150))
print(ph_b1,ph_b2)

ph_b=cmath.polar(ph_b1+ph_b2)
ph_b=(ph_b[0],math.degrees(ph_b[1]))
print('resultado: ', ph_b)

(4.286263797015736e-16+7j) (-2.598076211353316+1.4999999999999998j)
resultado:  (8.888194417315589, 106.99608805717719)


c) $\mathrm{V}_1 = 70 \angle{30^{\circ}} \: [\mathrm{V}] $ --->
$70\cos(\omega t+30^{\circ})$

d) $\mathrm{V}_2 =-60+\mathrm{j}40 \: [\mathrm{V}] $

In [6]:
ph_d=cmath.polar(-60+40j)
ph_d=(ph_d[0],math.degrees(ph_d[1]))
print(ph_d)

(72.11102550927978, 146.30993247402023)


---> $72.11\cos(\omega t+146.31^{\circ})$

___
#### 5- En el circuito de la figura:

<img src="c5_1.png" alt= “” width="40%" height="40%">

#### a) Si $\textbf{I}_{10}=2\angle{42^{\circ}} \: \mathrm{mA}$ y $\textbf{V}=40\angle{132^{\circ}} \: \mathrm{mV}$, ¿cuál es probablemente el tipo de elemento conectado a la derecha de la resistencia de 10 $\Omega$? Calcule su valor suponiendo que la fuente de tensión opera a una frecuencia de 1 000 rad/s.

In [7]:
I_10=cmath.rect(2e-3,math.radians(42))
V=cmath.rect(40e-3,math.radians(132))

Z=V/I_10
Zp=cmath.polar(Z)
print('Z_v =',Z,',',Zp[0],math.degrees(Zp[1]))
w=1000
L=np.abs(Z)/(w)
print('L =',np.round(L*1e3),'mH')

Z_v = (1.2891507748063227e-15+19.999999999999996j) , 19.999999999999996 90.0
L = 20.0 mH


#### b) Si $\textbf{I}_{10}=4\angle{35^{\circ}} \: \mathrm{A}$, $\textbf{V}=10\angle{35^{\circ}} \: \mathrm{V}$ e $\textbf{I}=2\angle{35^{\circ}} \: \mathrm{A}$, diga que tipo de elemento corresponde a la tensión $\textbf{V}$ y su valor. Además, determine $\textbf{V}_s$.

In [8]:
I_10=cmath.rect(4,math.radians(35))
V=cmath.rect(10,math.radians(35))
I=cmath.rect(2,math.radians(35))

Z=V/I_10
Zp=cmath.polar(Z)
print('Z_v =',Z,',',Zp[0],np.round(math.degrees(Zp[1])))
print('R =',Zp[0],'ohm')

I_25=I_10-I
Vs=cmath.polar(10*I_10+V+I_25*25)
print('Vs =',np.round(Vs[0]),np.round(math.degrees(Vs[1])))

Z_v = (2.5+1.8188829204768093e-16j) , 2.5 0.0
R = 2.5 ohm
Vs = 100.0 35.0


___
#### 6- Utilice las técnicas de análisis fasorial para obtener expresiones para las dos corrientes de malla $i_1$ e $i_2$ como se muestra en la figura.

<img src="c6.png" alt= “” width="60%" height="60%">

#### Utilizando mallas:

$-2.5\angle9+2I_1+j\omega L(I_1-I_2)$

$j\omega L(I_2-I_1)+\dfrac{I_2}{j\omega C}+5I_1=0$

In [9]:
w=10
L=1
C=330*1e-3
vs=cmath.rect(2.5,math.radians(9))

a=[[2+1j*w*L,-1j*w*L],[-1j*w*L+5,1j*w*L+1/(1j*w*C)]]
b=[vs,0]
I = np.linalg.solve(a, b)
I1=cmath.polar(I[0])
I2=cmath.polar(I[1])
print('i1 = ',np.round(I1[0],4)*1e3,'cos( 10t +',np.round(math.degrees(I1[1]),2),')','mA')
print('i2 = ',np.round(I2[0],4)*1e3,'cos( 10t +',np.round(math.degrees(I2[1]),2),')','mA')

i1 =  349.0 cos( 10t + 11.5 ) mA
i2 =  402.4 cos( 10t + 38.07 ) mA


#### Utilizando nodos:

$\dfrac{V-Vs}{2}+\dfrac{V}{j\omega L}+\dfrac{V-5I_1}{\dfrac{1}{j\omega C}}=0$

$I_1=\dfrac{Vs-V}{2}$

In [10]:
a=[[1/2+1/(1j*w*L)+1j*w*C,-5*1j*w*C],
   [1/2,1]]
b=[vs/2,vs/2]
S = np.linalg.solve(a, b)
I1=cmath.polar(S[1])
print('i1 = ',np.round(I1[0],4)*1e3,'cos( 10t +',np.round(math.degrees(I1[1]),2),')','mA')

I2=cmath.polar(S[0]/(1j*w*L)-S[1])
print('i1 = ',np.round(I2[0],4)*1e3,'cos( 10t +',np.round(math.degrees(I2[1]),2),')','mA')

i1 =  349.0 cos( 10t + 11.5 ) mA
i1 =  402.4 cos( 10t + -141.93 ) mA
